packages/react-aria-components/src/DropZone.tsx

```diff
@@ -0,0 +1,108 @@
+import {ContextValue, Provider, RenderProps, SlotProps, useContextProps, useRenderProps, useSlot} from './utils';
+import {DropOptions, mergeProps, useClipboard, useDrop, useFocusRing, useHover, useId, VisuallyHidden} from 'react-aria';
+import {FileTriggerContext} from './FileTrigger';
+import {filterDOMProps, useLabels} from '@react-aria/utils';
+import React, {createContext, ForwardedRef, forwardRef, useRef} from 'react';
+import {TextContext} from './Text';
+
+export interface DropZoneRenderProps {
+  /**
+   * Whether the dropzone is currently hovered with a mouse.
+   * @selector [data-hovered]
+   */
+  isHovered: boolean,
+  /**
+   * Whether the dropzone is focused, either via a mouse or keyboard.
+   * @selector :focus
+   */
+  isFocused: boolean,
+  /**
+   * Whether the dropzone is keyboard focused.
+   * @selector [data-focus-visible]
+   */
+  isFocusVisible: boolean,
+  /**
+   * Whether the dropzone is the drop target.
+   * @selector [data-drop-target]
+   */
+  isDropTarget: boolean
+}
+// note: possibly add isDisabled prop in the future
+export interface DropZoneProps extends Omit<DropOptions, 'getDropOperationForPoint' | 'onInsert' | 'onRootDrop' | 'onItemDrop' | 'onReorder'>, RenderProps<DropZoneRenderProps>, SlotProps, AriaLabelingProps {}
+
+export const DropZoneContext = createContext<ContextValue<DropZoneProps, HTMLDivElement>>(null);
+
+function DropZone(props: DropZoneProps, ref: ForwardedRef<HTMLDivElement>) {
+  [props, ref] = useContextProps(props, ref, DropZoneContext);
+  let buttonRef = useRef<HTMLButtonElement>(null);
+  let {dropProps, dropButtonProps, isDropTarget} = useDrop({...props, ref, buttonRef, hasDropButton: true});
+  let {hoverProps, isHovered} = useHover({});
+  let {focusProps, isFocused, isFocusVisible} = useFocusRing();
+  let [fileTriggerRef] = useSlot(); 
+  
+  let textId = useId();
+  let labelProps = useLabels({'aria-labelledby': textId});
+
+  let {clipboardProps} = useClipboard({
+    onPaste: (items) => props.onDrop?.({  
+      type: 'drop',
+      items,
+      x: 0,
+      y: 0,
+      dropOperation: 'copy'
+    })
+  });
+
+  let renderProps = useRenderProps({ 
+    ...props,
+    values: {isHovered, isFocused, isFocusVisible, isDropTarget},
+    defaultClassName: 'react-aria-DropZone'
+  });
+  let DOMProps = filterDOMProps(props);
+  delete DOMProps.id;
+  
+  return (
+    <Provider
+      values={[
+        [FileTriggerContext, {ref: fileTriggerRef}],
+        [TextContext, {id: textId, slot: 'heading'}]
+      ]}>
+      <div
+        {...mergeProps(dropProps, hoverProps, DOMProps)} 
+        {...renderProps}
+        ref={ref}
+        slot={props.slot} 
+        // onClick={() => buttonRef.current?.focus()}
+        data-hovered={isHovered || undefined}
+        data-focused={isFocused || undefined}
+        data-focus-visible={isFocusVisible || undefined}
+        data-drop-target={isDropTarget || undefined} >
+        <VisuallyHidden>
```

Comment:
since this is no longer conditional, if the user does include a FileTrigger, they first will first have to tab to the DropZone before they can tab to the FileTrigger. the reason for having it be conditional before was so that if a user did include a FileTrigger, they could just directly tab to the FileTrigger. not sure which behavior is more appropriate in this case.

Reply:
Just tested it, noticed that the visually hidden button is getting focus first hence the need for a second tab. Can we make the visually hidden button have tabIndex=-1 or something to prevent this from happening? Or does the visually hidden button need to be tabbable at all times?